base/repl/LineEdit.jl

```diff
@@ -182,8 +183,10 @@ function _clear_input_area(terminal, state::InputAreaState)
     clear_line(terminal)
 end
 
-prompt_string(s::PromptState) = s.p.prompt
+prompt_string(s::PromptState) = prompt_string(s.p)
+prompt_string(p::Prompt) = prompt_string(p.prompt)
 prompt_string(s::AbstractString) = s
+prompt_string(f::Function) = eval(Expr(:call, f))
```

Comment:
Yes I had world errors, so I did as in #19924. I'd rather keep it as is to be consistent with other such `eval`s in this file.